src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -312,24 +327,43 @@ def _forward(self, model_inputs):
             out = {"tokens": tokens}
         else:
             stride = model_inputs.pop("stride", None)
-            input_values = model_inputs.pop("input_values")
+            input_values = (
+                model_inputs.pop("input_values")
+                if "input_values" in model_inputs
+                else model_inputs.pop("input_features")
+            )
             attention_mask = model_inputs.pop("attention_mask", None)
-            outputs = self.model(input_values=input_values, attention_mask=attention_mask)
+            outputs = self.model(input_values, attention_mask=attention_mask)
             logits = outputs.logits
 
             if self.type == "ctc_with_lm":
                 out = {"logits": logits}
             else:
                 out = {"tokens": logits.argmax(dim=-1)}
             if stride is not None:
```

Comment:
Here is the main change. The previous approach of calculating the `inputs_to_logits_ratio` from the convolutional layers of the model is still used when possible. For M-CTC-T, essentially the new approach is:

1. Calculate the ratio of the original length of the input features (`stride[0]`) divided by the length of the output logits (`outputs.logits.shape[1]`)
2. Assume that this ratio holds for the strides